src/jupyterhub_cost_monitoring/query_usage.py

```diff
@@ -61,6 +69,10 @@ def query_usage(
         response = query_prometheus(USAGE_MAP[component_name], from_date, to_date)
         result.extend(_process_response(response, component_name))
     result = _filter_json(result, hub=hub_name, user=user_name)
+    # Calculate daily cost factors from absolute usage totals
+    result = _calculate_daily_cost_factors(result, hub_name=hub_name)
+    # sort the result by date
+    result.sort(key=lambda x: (x["date"], x["component"], x["hub"], x["user"]))
```

Comment:
```suggestion
    # Calculate daily cost factors from absolute usage totals
    result = _calculate_daily_cost_factors(result, hub_name=hub_name)
    # sort the result by date
    result.sort(key=lambda x: (x["date"], x["component"], x["hub"], x["user"]))
    result = _filter_json(result, hub=hub_name, user=user_name)
```

Apply the filters last. We have an issue in `_calculate_daily_cost_factors` where if I go to `/costs-per-user?from=2025-08-13T00:00Z&to=2025-08-13T23:59Z&component=compute&hub=workshop&user=*****` (which will happen in the Grafana), then the daily cost factor for that user is equal to 1.0. Let's get this PR merged first, and then create an issue to deal with that later.

Reply:
This should now be fixed through https://github.com/2i2c-org/jupyterhub-cost-monitoring/pull/7/commits/f6c827ffee3c0eb45c46a0ab59fb089622ccd4bd